### PR TITLE
Refactor code validators

### PR DESCRIPTION
**Pull request by shaharbar1**
Change log:
 1. Updated pydantic to 2.8.2
 2. Migrated base.py, cmab.py, smab.py, model.py, strategy.py based on new pydantic version.
 3. Added utils.py with utility function to convert dict to dict with json seralizable values.
 4. Changed relevant "get_state" UTs to use the new utility function.

### CODE REVIEW

File: pybandits/simulation_cmab.py [hunk 1]
```diff
@@ -261,7 +262,7 @@ def get_proportion_positive_reward(self):
             + str(i): (
                 self.results["action"].loc[(self.results["group"] == i) & (self.results["reward"] == 1)].value_counts()
                 / self.results["action"].loc[(self.results["group"] == i)].value_counts()
-            ).to_dict()
+            ).to_serializable_dict()
```
Comment (dariodandrea) [inline pybandits/simulation_cmab.py:265]: What was the issue here?

Comment (shaharbar1) [reply pybandits/simulation_cmab.py:265]: No bug. My bad as I refactored (to_serializable_dict was to_dict before I pushed the code).

File: pybandits/model.py [hunk 2]
```diff
@@ -60,11 +60,11 @@ class BaseBeta(Model):
...
-    @root_validator(pre=True)
+    @model_validator(mode="before")
     @classmethod
     def both_or_neither_counters_are_defined(cls, values):
-        n_successes_defined = "n_successes" in values.keys()
-        n_failures_defined = "n_failures" in values.keys()
+        n_successes_defined = hasattr(values, "n_successes")
+        n_failures_defined = hasattr(values, "n_failures")
         if n_successes_defined != n_failures_defined:
             raise ValueError("Either both or neither n_successes and n_failures should be specified.")
```
Comment (adarmiento) [inline pybandits/model.py:69]:
```suggestion
if hasattr(values, "n_successes") ^ hasattr(values, "n_failures"):
    raise ValueError("Either both or neither n_successes and n_failures should be specified.")
```

Comment (shaharbar1) [reply pybandits/model.py:69]: Thanks Alessandro

File: pybandits/base.py [hunk 3]
```diff
@@ -93,29 +89,31 @@ class BaseMab(PyBanditsBaseModel, ABC):
         The list of possible actions, and their associated Model.
     strategy: Strategy
         The strategy used to select actions.
-    epsilon: Optional[Float01], defaults to None
+    epsilon: Optional[Float01]
         The probability of selecting a random action.
+    default_action: Optional[ActionId]
+        Default action to select if the epsilon-greedy approach is used. None for random selection.
```
Comment (dariodandrea) [inline pybandits/base.py:95]: Suggestion to make it more verbose:
```
The default action to select with a probability of epsilon when using the epsilon-greedy approach. If `default_action` is None, a random action from the action set will be selected with a probability of epsilon.
```

**Review by dariodandrea — APPROVED** [review]
Good job @shaharbar1. I just left a minor comment for the docstring

File: pybandits/base.py (hunk 3 above)
Comment (shaharbar1) [reply pybandits/base.py]: Thanks. I also changed the docstring on smab.py and cmab.py.

**Pull request merged**